src/Core/Domain/Alias/Exception/AliasConstraintException.php

```diff
@@ -50,4 +50,9 @@ class AliasConstraintException extends AliasException
      * When alias visibility value is invalid
      */
     public const INVALID_VISIBILITY = 40;
+
+    /**
+     * Search limit must be a positive integer or null
+     */
+    public const INVALID_SEARCH_LIMIT = 50;
```

Comment:
im not sure this kind of input needs a validation. Doesn't bring much value, especially because it doesn't write to db, so I think null can be treated same as 0 and negative values can be casted to positive (you can search for examples of such queries in product domain, I think we never overcomplicated these things)